lib/util/yaml.ts

```diff
@@ -117,6 +125,29 @@ export function parseSingleYaml<ResT = unknown>(
   content: string,
   options?: YamlOptions<ResT>,
 ): ResT {
+  const rawDocument = parseSingleYamlDocument(content, options);
+
+  const document = rawDocument.toJS({ maxAliasCount: 10000 });
+  const schema = options?.customSchema;
+  if (!schema) {
+    return document as ResT;
+  }
+
+  return schema.parse(document);
+}
+
+/**
+ * Parse a YAML string into a Document representation.
+ *
+ * Only a single document is supported.
+ *
+ * @param content
+ * @param options
+ */
+export function parseSingleYamlDocument(
+  content: string,
+  options?: YamlParseDocumentOptions,
+): Document {
```

Comment:
I added this just to keep the parsing centralised, even if there are no other use-cases for YAML Document transforms at the moment.